File: 229-majority-element-ii/majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length;
        int t = n / 3;
        List<Integer> ans = new ArrayList<>();
        Map<Integer, Integer> my = new HashMap<>();

        for (int i : nums) {
            my.put(i, my.getOrDefault(i, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : my.entrySet()) {
            if (entry.getValue() > t) {
                ans.add(entry.getKey());
            }
        }

        return ans;
        
    }
}

File: 229-majority-element-ii/majority-element-ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var majorityElement = function(nums) {
     let n = nums.length;
    let t = Math.floor(n / 3);
    let ans = [];
    let my = {};

    for (let i of nums) {
        my[i] = (my[i] || 0) + 1;
    }

    for (let j in my) {
        if (my[j] > t) {
            ans.push(parseInt(j));
        }
    }

    return ans;
};